### PR TITLE
Fix queue message encoding for Azure (UTF-16 in XML)

### PR DESCRIPTION
**Pull request by cscharf**
Fixes https://app.asana.com/0/1169444489336079/1200549509965487/f.

Reverts #1425 

Based on discussions with @MGibson1  last week and needing to get something put together this week, taking a more simplistic approach that can be re-visited next week if necessary. This is _similar_ in approach to #1435 except that it moves the logic more lower-level in the queue service and alters all queue consumers with an extension method.

## Files
* **CoreHelpers.cs**: Adds `DecodeMessageText()` extension method to `QueueMessage` which does a test for Base64 and falls back to the raw message text. This was Matt's original idea before we started overcomplicating things.
* **AzureQueueService.cs**: 2 primary changes here, 1st is to always base64 encode the JSON message, the 2nd is to update the algorithm that determines and breaks up bulk enqueue requests by byte size to only serialize messages to JSON once, and to use the string byte-length overall to determine and calculate base64 size before creating finalized arrays in the message to return, and use string concatenation for the final JSON array (faster) vs. re-serializing to JSON (slower).
* **everywhere else**: Use the new `DecodeMessageText()` extension method instead of getting `QueueMessage.MessageText` directly.

## Testing Notes
* Noted in Asana and Slack

## Deployment Notes
**MUST deploy FIRST:**
   1. Event Processing Service
   2. Notification Service
   3. Admin Service
 
**THEN:**
   1. Identity
   2. API
   2. Anything else

### CODE REVIEW

File: src/Admin/HostedServices/AzureQueueBlockIpHostedService.cs [hunk 1]
```diff
@@ -35,7 +36,7 @@ protected override async Task ExecuteAsync(CancellationToken cancellationToken)
                     {
                         try
                         {
-                            await BlockIpAsync(message.MessageText, cancellationToken);
+                            await BlockIpAsync(message.DecodeMessageText(), cancellationToken);
```
Comment (MGibson1) [inline src/Admin/HostedServices/AzureQueueBlockIpHostedService.cs:39]: The service that queues these messages (`AzureQueueBlockIpService`) isn't backed by `AzureQueueService` so will not be sending encoded messages.

Currently `AzureQueueBlockIpService` just delivers ip addresses, so it'll constantly fail base64 decode operations.

Comment (cscharf) [reply src/Admin/HostedServices/AzureQueueBlockIpHostedService.cs]: Ah, when it checked references I could have sworn that came up, but it was when I was keying on `MessageText` itself so my wires got crossed, thanks Matt! Updated.

File: src/Admin/HostedServices/AzureQueueBlockIpHostedService.cs [hunk 2]
```diff
@@ -6,6 +6,7 @@
 using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
 using Azure.Storage.Queues;
+using Bit.Core.Utilities;
```
Comment (Hinton) [inline src/Admin/HostedServices/AzureQueueBlockIpHostedService.cs:9]: Is this needed?

File: src/Core/Services/Implementations/AzureQueueService.cs [hunk 3]
```diff
@@ -32,36 +34,62 @@ public async Task CreateManyAsync(IEnumerable<T> messages)
...
 
-            var messagesLists = new List<List<T>> { new List<T>() };
-            var strings = new List<string>();
-            var ListMessageLength = 2; // to account for json array brackets "[]"
-            foreach (var (message, jsonEvent) in messages.Select(m => (m, SerializeMessage(m))))
-            {
+            var messagesList = new List<string>();
+            var messagesListSize = 0;
+            
+            int calculateByteSize(int totalSize, int toAdd) =>
```
Comment (Hinton) [inline src/Core/Services/Implementations/AzureQueueService.cs:57]: Haven't looked into it in-depth. But there is a `Base64.GetMaxEncodedToUtf8Length` helper we could perhaps use instead? https://docs.microsoft.com/en-us/dotnet/api/system.buffers.text.base64.getmaxencodedtoutf8length?view=net-5.0#system-buffers-text-base64-getmaxencodedtoutf8length(system-int32)

File: src/Admin/HostedServices/AzureQueueBlockIpHostedService.cs (hunk 2 above)
Comment (cscharf) [reply src/Admin/HostedServices/AzureQueueBlockIpHostedService.cs:9]: No, good catch.

**Review by Hinton — DISMISSED** [review]
Looks good, two minor comments.

File: src/Core/Services/Implementations/AzureQueueService.cs (hunk 3 above)
Comment (cscharf) [reply src/Core/Services/Implementations/AzureQueueService.cs:57]: Hi @Hinton , I missed that, however from the [src of that method](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Memory/src/System/Buffers/Text/Base64Encoder.cs#L155) it looks like my implementation here is more accurate and most of the logic is still necessary, including this particular method that determines the total combined "next" JSON block to be added to see if it needs to start a new batch. I think we're good on this one IMHO.

**Review by Hinton — APPROVED** [review]

**Pull request merged**